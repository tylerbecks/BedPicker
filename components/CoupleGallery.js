import React from "react";
import { Image, StyleSheet, View, Text } from "react-native";

const AVATAR_SIZE = 40;

const styles = StyleSheet.create({
  container: {
    flexDirection: "row",
    marginBottom: 20,
    flexWrap: "wrap"
  },
  couple: {
    alignItems: "center",
    flexDirection: "row",
    margin: 10
  },
  circle: {
    alignItems: "center",
    backgroundColor: "#CFD8DC",
    borderColor: "black",
    borderRadius: AVATAR_SIZE / 2,
    borderWidth: 2,
    height: AVATAR_SIZE,
    width: AVATAR_SIZE
  },
  firstImage: {
    zIndex: 2
  },
  secondImage: {
    marginLeft: -7,
    zIndex: 1
  },
  textContainer: {
    backgroundColor: "#2196F3",
    justifyContent: "center"
  },
  text: {
    color: "white",
    fontSize: 14,
    fontWeight: "bold",
    textAlign: "center"
  }
});

const getFirstInitial = name => name.charAt(0);

const CoupleGallery = ({ couples }) => (
  <View style={styles.container}>
    {couples.map(([partner1, partner2]) => (
      <View style={styles.couple} key={partner1.id}>
        <ImageOrTextCircle
          style={{ ...styles.circle, ...styles.firstImage }}
          guest={partner1}
        />
        <ImageOrTextCircle
          style={{ ...styles.circle, ...styles.secondImage }}
          guest={partner2}
        />
      </View>
    ))}
  </View>
);

const ImageOrTextCircle = ({ style, guest }) =>
  guest.photo ? (
    <Image style={style} source={guest.photo} />
  ) : (
    <View style={{ ...style, ...styles.textContainer }}>
      <Text style={styles.text}>{getFirstInitial(guest.name)}</Text>
    </View>
  );

export default CoupleGallery;
