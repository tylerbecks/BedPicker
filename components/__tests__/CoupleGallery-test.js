import 'react-native';
import React from 'react';
import CoupleGallery from '../CoupleGallery';
import renderer from 'react-test-renderer';

it('renders', () => {
  const guest1 = { id: 1, name: 'foo' };
  const guest2 = { id: 2, name: 'bar' };
  expect(
    renderer.create(<CoupleGallery couples={[[guest1, guest2]]} />).toJSON()
  ).toMatchSnapshot();
});
