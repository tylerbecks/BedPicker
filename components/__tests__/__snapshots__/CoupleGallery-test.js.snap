// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "flexWrap": "wrap",
      "marginBottom": 20,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "margin": 10,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#2196F3",
          "borderColor": "black",
          "borderRadius": 20,
          "borderWidth": 2,
          "height": 40,
          "justifyContent": "center",
          "width": 40,
          "zIndex": 2,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 14,
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        f
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#2196F3",
          "borderColor": "black",
          "borderRadius": 20,
          "borderWidth": 2,
          "height": 40,
          "justifyContent": "center",
          "marginLeft": -7,
          "width": 40,
          "zIndex": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 14,
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        b
      </Text>
    </View>
  </View>
</View>
`;
